File: Assets/Scripts/AISimple.js
#pragma strict

private var HitPoint: RaycastHit;
private var MoveTo = 1;
private var One = 1;
public var MoveSpeed = 0.01;
public var MaxSpeed = 7;

function Start () {
        MoveTo = MoveSpeed;
}

function FixedUpdate () {
        if(Mathf.Abs(rigidbody.velocity.x) < MaxSpeed)
                rigidbody.AddForce (MoveTo, 0, 0);
        var PosTMP = Vector3(this.transform.position.x + One, this.transform.position.y, this.transform.position.z);
        Debug.DrawLine(this.transform.position, PosTMP);
        if (!Physics.Raycast(PosTMP, Vector3.down, HitPoint, 1)) {
                if(MoveTo > 0){
                        MoveTo = -MoveSpeed;
                        One = -1;
                }
                else{
                        MoveTo = MoveSpeed;
                        One = 1;
                }
        }
}

File: Assets/Scripts/AISimple.cs
// This file is part of HAT
// 
// Copyright (c) 2016 sietze greydanus
// 
// HAT is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License version 3, as
// published by the Free Software Foundation.
// 
// HAT is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
// 
// You should have received a copy of the GNU General Public License
// along with HAT. If not, see <http://www.gnu.org/licenses/>.
// 

using UnityEngine;

public class AISimple : MonoBehaviour
{
    private float _moveTo = 1;
    private int _one = 1;
    public float MaxSpeed = 7;
    public float MoveSpeed = 0.01f;

    private void Start()
    {
        _moveTo = MoveSpeed;
    }

    private void FixedUpdate()
    {
        if (Mathf.Abs(GetComponent<Rigidbody>().velocity.x) < MaxSpeed)
            GetComponent<Rigidbody>().AddForce(_moveTo, 0, 0);
        var PosTMP = new Vector3(transform.position.x + _one, transform.position.y, transform.position.z);
        Debug.DrawLine(transform.position, PosTMP);
        if (!Physics.Raycast(PosTMP, Vector3.down, 1f))
        {
            if (_moveTo > 0)
            {
                _moveTo = -MoveSpeed;
                _one = -1;
            }
            else
            {
                _moveTo = MoveSpeed;
                _one = 1;
            }
        }
    }
}